rkt/list.go

```diff
@@ -180,12 +198,31 @@ func runList(cmd *cobra.Command, args []string) int {
 		return 1
 	}
 
+	switch flagFormat {
+	case "json":
+		result, err := json.Marshal(pods)
+		if err != nil {
+			stderr.PrintE("error marshaling the pods", err)
+			return 1
+		}
+		stdout.Print(string(result))
+	case "json-pretty":
+		result, err := json.MarshalIndent(pods, "", "\t")
+		if err != nil {
+			stderr.PrintE("error marshaling the pods", err)
+			return 1
+		}
+		stdout.Print(string(result))
+	}
+
 	if len(errors) > 0 {
 		printErrors(errors, "listing pods")
 	}
 
-	tabOut.Flush()
-	stdout.Print(tabBuffer)
+	if flagFormat == "" {
```

Comment:
@lucab Then the flush is performed before the errors are printed, which changed the behavior. I can do that if you think that's fine.